osutil/disks/disks_linux.go

```diff
@@ -188,6 +188,24 @@ var diskFromDeviceName = func(deviceName string) (Disk, error) {
 	return diskFromUdevProps(deviceName, "name", props)
 }
 
+func rootMountPointsForPartition(part Partition) ([]string, error) {
+	mounts, err := osutil.LoadMountInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	mountpoints := []string{}
+	for _, mnt := range mounts {
+		if mnt.DevMajor == part.Major && mnt.DevMinor == part.Minor {
+			if mnt.Root == "/" {
+				mountpoints = append(mountpoints, mnt.MountDir)
```

Comment:
Do you already have a use case for knowing whether the mount point is writable?

Reply:
Actually tbh I thought we didn't need to but actually the /var/lib/snapd/seed mountpoint is actually read-only from my paste above to mvo, so I think in my testing so far I just happened to not run into the problem where a mountpoint of the partition I am looking for to update, so I think I need to add this before we enable the feature fully since it is a possibility that we could be trying to update files on the ubuntu-seed filesystem (i.e. on raspi) and so we need to find the mountpoint of that structure and here we have one that could be read-only.

Reply:
Done in https://github.com/snapcore/snapd/pull/10872